src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs

```diff
@@ -213,7 +214,7 @@ public IWebHostBuilder UseStartup(Type startupType)
             return this;
         }
 
-        public IWebHostBuilder UseStartup(Func<WebHostBuilderContext, object> startupFactory)
+        public IWebHostBuilder UseStartup<TStartup>(Func<WebHostBuilderContext, TStartup> startupFactory)
```

Comment:
Dumb question but why is this better? Wouldn't the startup type be referenced in the func when it is created?

Reply:
I think I know. It's because we want to ensure Configure/ConfigureServices as included. Creating the type in the func will ensure the constructor is available, but the methods will get trimmed because they're not used.

A generic type will hit the method with an attribute on the generic type arg that ensures methods are preserved.

Reply:
Yep